### DISCUSSION TITLE
Retaining current core framework?

### DISCUSSION THREAD

**Discussion by AS-EBaiden** [storybookjs/storybook#25462] [Help]
### Summary

Hello all, I've been attempting to upgrade SB 6.5 to the latest version of 7+ for going on 5 days now, and I'm ready to throw in the towel, but before I do, maybe you can spot the things I'm not doing right. 
Im working with a monorepo.
I was given the prompt `npx storybook@latest upgrade`. I noticed that it kept detecting next.js and it was instructing me to 
remove:
 * @storybook/builder-webpack5│   
 * @storybook/manager-webpack5│   
 
 and  Add the following dependencies:│   
* @storybook/nextjs

I'm not ready to use nextjs yet, and I'd like to retain just react, would that be possible?

I thought I'd give it a try and skip the `new-framework migration` in the prompt, and after selecting most of the defaults this is what i was displayed.
![image](https://github.com/storybookjs/storybook/assets/110992392/8b148a8b-8b1a-4624-8567-55a55f7d7159)
The incompatible-addons is @storybook/addon-postcss@2.0.0 (so I should remove that right, or is there another version i can replace it with.) 

 and after doing that per the instruction on the migration documentation I ran an `pnpm dlx storybook@latest init` which is where im seeing all the errors. 
I have @storybook/addon-postcss@2.0.0, and it's saying it's no longer supported (so I should remove that right?) 

also this is what my main.js is looking like
```js
import { dirname, join } from "path";
const excludedProps = ['theme', 'as', 'forwardedAs', 'gradient', 'typeStyle', 'ref']

const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin')

module.exports = {
  staticDirs: ['../public'],
  stories: ['../src/**/*.stories.mdx'],
  framework: getAbsolutePath("@storybook/react"),

  core: {
    builder: getAbsolutePath("@storybook/builder-webpack5"),
  },

  features: {
    babelModeV7: true,

    /**
     * Webpack 5: This doesn't work because of mixing import/require statements in (I think) Affinity & theme source code.
     */
    // storyStoreV7: true,
  },

  addons: [/**
   * There is a very specific set of dependencies required for this to work.
   * Both `react-refresh` resolutions in the root package.json are important.
   */
  // '@storybook/preset-create-react-app',

  /**
   * Don't use this in conjunction with `preset-create-react-app`.
   * They both add their own CSS loaders which conflict.
   * See https://github.com/storybookjs/storybook/issues/9016 for an explanation.
   */

  {
    name: '@storybook/addon-essentials',
    options: {
      // backgrounds: false,
    },
  }, getAbsolutePath("@storybook/addon-a11y"), getAbsolutePath("@storybook/addon-mdx-gfm")],

  reactOptions: {
    /**
     * Couldn't enable due to https://github.com/storybookjs/storybook/issues/15223
     */

  },

  typescript: {
    check: true,
    checkOptions: {},
    reactDocgen: 'react-docgen-typescript',

    /**
     * These options improve ArgsTable generation.
     */
    reactDocgenTypescriptOptions: {
      savePropValueAsString: true,
      shouldExtractLiteralValuesFromEnum: true,
      shouldRemoveUndefinedFromOptional: true,
      propFilter: (prop) =>
        (prop.parent ? !/node_modules/.test(prop.parent.fileName) : true) && excludedProps.indexOf(prop.name) < 0,
    },
  },

  webpackFinal: async (config) => {
    config.resolve.plugins = [
      ...(config.resolve.plugins || []),
      new TsconfigPathsPlugin({
        extensions: config.resolve.extensions,
      }),
    ]

    return config
  },

  docs: {
    autodocs: true
  }
}

function getAbsolutePath(value) {
  return dirname(require.resolve(join(value, "package.json")));
}
```

**Comment by Marklb** [comment]
I don't use React, so I am not the most experienced, but I can maybe provide some hints. If you can't share your project, for someone to look at the full config, maybe you could clone it and just remove everything other than maybe a small example component that gives an idea of what you are doing. Someone not too familiar with React projects, but familiar with Storybook internals, like myself, could probably work through migrating config.

If you can't share a project, the following are my thoughts that may help point out problems, but maybe someone that uses React can correct or clarify any of my suggestions:

- You are correct about removing `@storybook/addon-postcss`, but without seeing your project it is difficult to say how to replace it. `@storybook/addon-postcss` was last updated 3 years ago and replaced by `@storybook/addon-styling`, which is also now deprecated and replaced by to packages depending on what you need. The readme for [`@storybook/addon-styling`](https://storybook.js.org/addons/@storybook/addon-styling) briefly explains what the two replacements, `@storybook/addon-styling-webpack` and `@storybook/addon-themes`, are for. Migrating from `@storybook/addon-postcss`, you probably only need `@storybook/addon-styling-webpack`. You may not actually need either, but without seeing more of your project I can't be sure.
- `@storybook/manager-webpack5` only has a v7 alpha release, so you would want to remove that one.
- Instead of putting the `core: { builder: "@storybook/builder-webpack5" }` in your `.storybook/main.js`, maybe replace the framework with `framework: "@storybook/react-webpack5"`.
- You didn't specifically mention what you use, instead of nextjs, but if it is unnecessarily detecting nextjs then maybe you could temporarily remove the `next.js` dependency or config that is being detected. I don't really know anything about nextjs, so I would be able to step through the code and see why it is recommending that, but without seeing the project structure and config that is the only suggestion I have.
